editor/gui/editor_scene_tabs.cpp

```diff
@@ -217,8 +217,12 @@ void EditorSceneTabs::update_scene_tabs() {
 	Ref<Texture2D> script_icon = get_editor_theme_icon(SNAME("Script"));
 	for (int i = 0; i < EditorNode::get_editor_data().get_edited_scene_count(); i++) {
 		Node *type_node = EditorNode::get_editor_data().get_edited_scene_root(i);
+
+		// We should not call get_object_icon if the scene we are working on is a removing scene, since this will populate _script_icon_cache with script reference we just cleared
```

Comment:
```suggestion
		// We should not call get_object_icon if the scene we are working on is a removing scene,
		// since this will populate _script_icon_cache with script reference we just cleared.
```